elasticdl/python/elasticdl/feature_column/feature_column.py

```diff
@@ -0,0 +1,156 @@
+import collections
+import math
+
+import tensorflow as tf
+from tensorflow.python.feature_column import feature_column as fc_old
+from tensorflow.python.feature_column import feature_column_v2 as fc_lib
+from tensorflow.python.framework import tensor_shape
+from tensorflow.python.ops import init_ops
+
+
+def embedding_column(
+    categorical_column,
+    dimension,
+    combiner="mean",
+    initializer=None,
+    ckpt_to_load_from=None,
+    tensor_name_in_ckpt=None,
+    max_norm=None,
+    trainable=True,
+):
+    if (dimension is None) or (dimension < 1):
+        raise ValueError("Invalid dimension {}.".format(dimension))
+    if (ckpt_to_load_from is None) != (tensor_name_in_ckpt is None):
+        raise ValueError(
+            "Must specify both `ckpt_to_load_from` and "
+            "`tensor_name_in_ckpt` or none of them."
+        )
+
+    if (initializer is not None) and (not callable(initializer)):
+        raise ValueError(
+            "initializer must be callable if specified. "
+            "Embedding of column_name: {}".format(categorical_column.name)
+        )
+    if initializer is None:
+        initializer = init_ops.truncated_normal_initializer(
+            mean=0.0, stddev=1 / math.sqrt(dimension)
+        )
+
+    return EmbeddingColumn(
+        categorical_column=categorical_column,
+        dimension=dimension,
+        combiner=combiner,
+        initializer=initializer,
+        ckpt_to_load_from=ckpt_to_load_from,
+        tensor_name_in_ckpt=tensor_name_in_ckpt,
+        max_norm=max_norm,
+        trainable=trainable,
+    )
+
+
+class EmbeddingColumn(
+    fc_lib.DenseColumn,
+    fc_lib.SequenceDenseColumn,
+    fc_old._DenseColumn,
+    fc_old._SequenceDenseColumn,
+    collections.namedtuple(
+        "EmbeddingColumn",
+        (
+            "categorical_column",
+            "dimension",
+            "combiner",
+            "initializer",
+            "ckpt_to_load_from",
+            "tensor_name_in_ckpt",
+            "max_norm",
+            "trainable",
+        ),
+    ),
+):
+    def __init__(self, **kwargs):
+        self.tape = None
+
+    @property
+    def _is_v2_column(self):
+        return (
+            isinstance(self.categorical_column, fc_lib.FeatureColumn)
+            and self.categorical_column._is_v2_column
+        )
+
+    @property
+    def name(self):
+        """See `FeatureColumn` base class."""
+        return "{}_embedding_elasticdl".format(self.categorical_column.name)
+
+    @property
+    def parse_example_spec(self):
+        """See `FeatureColumn` base class."""
+        return self.categorical_column.parse_example_spec
+
+    @property
+    def variable_shape(self):
+        """See `DenseColumn` base class."""
+        return tensor_shape.TensorShape([self.dimension])
+
+    def create_state(self, state_manager):
```

Comment:
What's the use of the function?

Reply:
This function is just like the layer.build call. For native embedding column, it's used to create the embedding variables. But for the customized embedding column, we don't need create this variable. So we set the implementation to be empty.